prime-router/src/test/kotlin/fhirengine/azure/FHIRTranslatorIntegrationTests.kt

```diff
@@ -0,0 +1,364 @@
+                conditionFilter = listOf("true"),
+                format = MimeFormat.HL7
+            )
+        )
+        val receivers = UniversalPipelineTestUtils.createReceivers(receiverSetupData)
+        val org = UniversalPipelineTestUtils.createOrganizationWithReceivers(receivers)
+        val translator = createFHIRTranslator(azureEventService, org)
+        val reportContents = File(MULTIPLE_TARGETS_FHIR_PATH).readText()
+        val receiveReport = UniversalPipelineTestUtils.createReport(
+            reportContents,
+            TaskAction.receive,
+            Event.EventAction.CONVERT,
+            azuriteContainer
+        )
+        val translateReport = UniversalPipelineTestUtils.createReport(
+            MimeFormat.FHIR,
+            TaskAction.receive,
+            TaskAction.translate,
+            Event.EventAction.SEND,
+            Topic.FULL_ELR,
+            receiveReport,
+            BlobAccess.uploadBlob(
+                "${TaskAction.translate.literal}/mr_fhir_face.fhir",
+                reportContents.toByteArray(),
+                getBlobContainerMetadata(azuriteContainer)
+            )
+        )
+
+        val queueMessage = generateQueueMessage(
+            translateReport,
+            reportContents,
+            UniversalPipelineTestUtils.fhirSenderWithNoTransform,
+            "phd.x"
+        )
+        val fhirFunctions = UniversalPipelineTestUtils.createFHIRFunctionsInstance()
+
+        // execute
+        fhirFunctions.doTranslate(queueMessage, 1, translator)
+
+        // verify task and report_file tables were updated correctly in the Translate function (new task and new
+        // record file created)
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            val batchTask = DSL.using(txn).select(Task.TASK.asterisk()).from(Task.TASK)
+                .where(Task.TASK.NEXT_ACTION.eq(TaskAction.batch))
+                .fetchOneInto(Task.TASK)
+
+            // verify batch queue task exists
+            assertThat(batchTask).isNotNull()
+
+            val sendReportFile =
+                DSL.using(txn).select(ReportFile.REPORT_FILE.asterisk())
+                    .from(ReportFile.REPORT_FILE)
+                    .where(
+                        ReportFile.REPORT_FILE.REPORT_ID
+                            .eq(batchTask!!.reportId)
+                    )
+                    .fetchOneInto(ReportFile.REPORT_FILE)
+            assertThat(sendReportFile).isNotNull()
+
+            // verify message format is HL7
+            assertThat(batchTask.bodyFormat).isEqualTo("HL7")
+
+            // verify message does not match the original FHIR input
+            val translatedValue = BlobAccess.downloadBlobAsByteArray(
+                sendReportFile!!.bodyUrl,
+                UniversalPipelineTestUtils.getBlobContainerMetadata(azuriteContainer)
+            )
+            assertThat(translatedValue).isNotEqualTo(reportContents.toByteArray())
```

Comment:
Is there any way we could ensure enrichments were applied correctly? Perhaps apply them to the parsed original bundle and then assert with that?

Reply:
Looks like this is unresolved

Reply:
Discussed with Gabriel and added some clarifying comments on what this test is meant to accomplish.